sdl_android/src/main/java/com/smartdevicelink/api/VideoStreamingManager.java

```diff
@@ -0,0 +1,571 @@
+			return null;
+		}
+		IVideoStreamListener listener = internalInterface.startVideoStream(encrypted, parameters);
+		if(listener != null){
+			stateMachine.transitionToState(StreamingStateMachine.STARTED);
+		}else{
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+		}
+		return listener;
+	}
+
+	/**
+	 * Starts video service, sets up encoder, haptic manager, and remote display. Begins streaming the remote display.
+	 * @param parameters Video streaming parameters including: codec which will be used for streaming (currently, only
+	 *                    VideoStreamingCodec.H264 is accepted), height and width of the video in pixels.
+	 * @param encrypted Specify true if packets on this service have to be encrypted
+	 */
+	private void startVideoStreaming(VideoStreamingParameters parameters, boolean encrypted){
+		IVideoStreamListener streamListener = startVideoStream(parameters, encrypted);
+		if(streamListener == null){
+			Log.e(TAG, "Error starting video service");
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			return;
+		}
+		VideoStreamingCapability capability = (VideoStreamingCapability) internalInterface.getCapability(SystemCapabilityType.VIDEO_STREAMING);
+		if(capability != null && capability.getIsHapticSpatialDataSupported()){
+			hapticManager = new HapticInterfaceManager(internalInterface);
+		}
+		try {
+			encoder.init(this.context.get(),streamListener,parameters);
+			//We are all set so we can start streaming at at this point
+			encoder.start();
+			//Encoder should be up and running
+			createRemoteDisplay(encoder.getVirtualDisplay());
+			stateMachine.transitionToState(StreamingStateMachine.STARTED);
+		} catch (Exception e) {
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			e.printStackTrace();
+		}
+		Log.d(TAG, parameters.toString());
+	}
+
+	/**
+	 * Stops streaming service and remote display encoder if applicable.
+	 */
+	public void stopStreaming(){
+		if(remoteDisplay!=null){
+			remoteDisplay.stop();
+			remoteDisplay = null;
+		}
+		if(encoder!=null){
+			encoder.shutDown();
+		}
+		if(internalInterface!=null){
+			internalInterface.stopVideoService();
+		}
+		stateMachine.transitionToState(StreamingStateMachine.STOPPED);
+	}
+
+	/**
+	 * Stops video streaming service and removes service listener.
+	 */
+	public void dispose(){
+		stopStreaming();
+
+		// Remove listeners
+		internalInterface.removeServiceListener(SessionType.NAV, serviceListener);
+		internalInterface.removeOnRPCNotificationListener(FunctionID.ON_TOUCH_EVENT, touchListener);
+		internalInterface.removeOnRPCNotificationListener(FunctionID.ON_HMI_STATUS, hmiListener);
+
+		stateMachine.transitionToState(StreamingStateMachine.NONE);
+	}
+
+	// PUBLIC METHODS FOR CHECKING STATE
+
+	/**
+	 * Check if a video service is currently active
+	 * @return boolean (true = active, false = inactive)
+	 */
+	public boolean isVideoConnected(){
```

Comment:
I'm confused by the return here since it will return true even if the state is `STOPPED`. I think the naming could be better here as well, since this is already the video streaming manager we don't need to preface everything with video, could just be `isStreaming` or `isActive` but it will depend on what info it's intending to return.

Reply:
Maybe I should change the name to `isActive()`. I intended this for this to give developer an idea of whether the video service was started / is running successfully

Reply:
So there are a few different states and I think we just need to define what is important for the developer to know: 

1. Video manager is ready to be used
2. Video service has been started
3. Streaming has started and is active

So what does active mean? Maybe we have different methods to define those states. I'm open for suggestions here.